pymc/step_methods/hmc/nuts.py

```diff
@@ -296,7 +318,7 @@ def extend(self, direction):
         """
         if direction > 0:
             tree, diverging, turning = self._build_subtree(
-                self.right, self.depth, floatX(np.asarray(self.step_size))
+                self.right, self.depth, np.asarray(self.step_size, dtype=self.floatX)
```

Comment:
iirc if you do `np.asarray` on an array, numpy will copy and re-create the array. Is it necessary to do this at every step? Or can `step_size` be set just once once?

Reply:
np.asarray doesn't copy if it's already an array. np.array does